iroha/src/genesis.rs

```diff
@@ -0,0 +1,110 @@
+        init_config: &InitConfiguration,
+    ) -> Result<GenesisBlock, String> {
+        Ok(
+            GenesisBlock::load_genesis_file_from_path(&config.genesis_file_path, &init_config)
+                .expect("Failed to open a genesis block file"),
+        )
+    }
+
+    /// This method will build `Configuration` from a json *pretty* formatted file (without `:` in
+    /// key names).
+    /// # Panics
+    /// This method will panic if configuration file presented, but has incorrect scheme or format.
+    /// # Errors
+    /// This method will return error if system will fail to find a file or read it's content.
+    fn load_genesis_file_from_path<P: AsRef<Path> + Debug>(
+        path: P,
+        init_config: &InitConfiguration,
+    ) -> Result<GenesisBlock, String> {
+        let file =
+            File::open(path).map_err(|e| format!("Failed to open a genesis block file: {}", e))?;
+        let reader = BufReader::new(file);
+        let raw_block: RawGenesisBlock = serde_json::from_reader(reader)
+            .map_err(|e| format!("Failed to deserialize json from reader: {}", e))?;
+        let genesis_key_pair = KeyPair {
+            public_key: init_config.genesis_account_public_key.clone(),
+            private_key: init_config
+                .genesis_account_private_key
+                .clone()
+                .ok_or("genesis account private key is empty")?,
+        };
+        Ok(GenesisBlock {
+            txs: raw_block
+                .txs
+                .iter()
+                .map(|rtx| {
+                    Transaction::new(
+                        rtx.isi.clone(),
+                        <Account as Identifiable>::Id::genesis_account(),
+                        0,
+                    )
+                    .sign(&genesis_key_pair)
+                })
+                .map(|tx| tx?.accept())
+                .filter_map(Result::ok)
+                .collect(),
+        })
+    }
+}
+
+/// This module contains all configuration related logic.
+pub mod config {
+    use serde::Deserialize;
+    use std::env;
+
+    const GENESIS_FILE_PATH: &str = "GENESIS_FILE_PATH";
+    const DEFAULT_GENESIS_FILE_PATH: &str = "tests/genesis.json";
+
+    /// `GenesisConfiguration` provides an ability to define parameters such as `GENESIS`.
+    #[derive(Clone, Deserialize, Debug)]
+    #[serde(rename_all = "UPPERCASE")]
+    pub struct GenesisConfiguration {
+        /// Genesis block json path.
+        #[serde(default = "default_genesis_file_path")]
+        pub genesis_file_path: String,
+    }
+
+    impl GenesisConfiguration {
+        /// Load environment variables and replace predefined parameters with these variables
+        /// values.
+        pub fn load_environment(&mut self) -> Result<(), String> {
+            if let Ok(genesis_file_path) = env::var(GENESIS_FILE_PATH) {
+                self.genesis_file_path = genesis_file_path;
+            }
+            Ok(())
+        }
```

Comment:
We don't need to load genesis path from the environment, we simply tell in CLI if we want genesis to be loaded from path or not.

Reply:
ok!